test/cluster.test.ts

```diff
@@ -21,4 +21,11 @@ test('create the default cluster', () => {
     }})
 
   expect(stack).toHaveResource('AWS::AutoScaling::LaunchConfiguration')
-})
+});
+
+test('add s3 removalPolicy', () => {
+  const app = new App();
+  const stack = new Stack(app, 'testing-stack');
+  new k3s.Cluster(stack, 'Cluster-s3-removalPolicy')
+  expect(stack).toHaveResource('AWS::S3::Bucket');
+});
```

Comment:
I was wondering if we should introduce a new property in `ClusterProps` for example `bucketRemovalPolicy?: RemovalPolicy` and the default value is `RemovalPolicy.RETAIN`. When we specify the value as `RemovalPolicy.DESTROY`,  the custom resource removes the s3 bucket completely like this. WDYT @mreferre ? Does it make sense to retain the bucket as default behavior?

Reply:
I like the idea of adding a property to control this behavior. The bucket only contains the the k3s secret and the config.yaml to connect to the cluster and once the cluster is gone the value of those two files is close to 0. Having that said I don't like the idea to delete the bucket by default. I'd rather document front and center this property and call out that the default is to leave the bucket behind but users can opt in and decide to delete it if so they wish.